rust/private/clippy.bzl

```diff
@@ -145,6 +145,7 @@ rust_clippy_aspect = aspect(
             default = Label("@bazel_tools//tools/cpp:current_cc_toolchain"),
         ),
         "_error_format": attr.label(default = "//:error_format"),
+        "_extra_codegen": attr.label(default = "//:extra_codegen"),
```

Comment:
So it's quite likely we'll grow the number of these settings that each rule will want to depend on. What do you think about (all or subset of):
* Creating a `config` package at //rust/private/config
* There the BUILD file will define build_setting targets
* There the config.bzl file will define providers, rules, and a list of implicit attributes all the rules can just add
* rust_common struct will have a config sub-struct with functions returning the provider of the particular build_setting, something that instead of `ctx.attr._extra_codegen[ExtraCodegenInfo].extra_codegen` the user will write `rust_common.config.extra_codegen(ctx)`.

The goal is to allow third_party rules to depend on rust specific build settings without hardcoding the name and label of implicit attributes, and without needing to use the provider associated with the setting.

Reply:
I'm not sure I understand what you've outlined here. Is the goal that e.g. a `rust_binary` rule could ask for LTO and it would automatically apply down to all dependencies? That would be great, but I didn't think that was possible -- if it is I can certainly try to reimplement it that way.

Reply:
Nope nope I only outlined a way to structure the code in a way that protects third_party (for example if rules_ruby wanted to support linking native dependencies from rules_rust) rules from depending on internals of rules rust. Does it make sense? Which part doesn't?

Reply:
Ok -- I think I follow now, this is a code organization / api surface suggestion, right? I'll take a look.

Reply:
Sorry -- I wasn't able to get time to work on this. I've rebased the PR on top of the latest commits, but I don't think I have bandwidth for the refactor you're talking about right now.